crates/storage-pg/src/user/email.rs

```diff
@@ -191,6 +191,43 @@ impl UserEmailRepository for PgUserEmailRepository<'_> {
         Ok(Some(user_email.into()))
     }
 
+    #[tracing::instrument(
+        name = "db.user_email.find_by_email",
+        skip_all,
+        fields(
+            db.query.text,
+            user_email.email = email,
+        ),
+        err,
+    )]
+    async fn find_by_email(&mut self, email: &str) -> Result<Option<UserEmail>, Self::Error> {
+        let res = sqlx::query_as!(
+            UserEmailLookup,
+            r#"
+                SELECT user_email_id
+                     , user_id
+                     , email
+                     , created_at
+                FROM user_emails
+                WHERE email = $1
```

Comment:
That column isn't indexed, so we'll get poor performance in production. I've added an index for it in https://github.com/element-hq/matrix-authentication-service/pull/4385 though, so I'll merge this PR as-is

Reply:
Yes indeed. Again thank you for your help on this.